grr/server/grr_response_server/gui/ui/components/timestamp/timestamp_test.ts

```diff
@@ -0,0 +1,210 @@
+    expect(fixture.nativeElement.innerText).toEqual('1min ago');
+
+    componentInstance.date = new Date('2020-07-01T12:29:30.000');
+    fixture.detectChanges();
+    expect(fixture.nativeElement.innerText).toEqual('30min ago');
+
+    componentInstance.date = new Date('2020-07-01T12:00:01.001');
+    fixture.detectChanges();
+    expect(fixture.nativeElement.innerText).toEqual('59min ago');
+
+    componentInstance.date = new Date('2020-07-01T12:00:00.000');
+    fixture.detectChanges();
+    expect(fixture.nativeElement.innerText).not.toEqual('59min ago');
+    expect(fixture.nativeElement.innerText).not.toEqual('60min ago');
+  });
+
+  it('shows "XhYmin ago" for dates between 1 hour than 24 hours ago', () => {
+    const fixture = TestBed.createComponent(Timestamp);
+    const componentInstance = fixture.componentInstance;
+
+    componentInstance.date = new Date('2020-07-01T12:00:00.000');
+    fixture.detectChanges();
+    expect(fixture.nativeElement.innerText).toEqual('1h0min ago');
+
+    componentInstance.date = new Date('2020-07-01T01:25:00');
+    fixture.detectChanges();
+    expect(fixture.nativeElement.innerText).toEqual('11h35min ago');
+
+    componentInstance.date = new Date('2020-06-30T13:00:00.001');
+    fixture.detectChanges();
+    expect(fixture.nativeElement.innerText).toEqual('23h59min ago');
+
+    componentInstance.date = new Date('2020-06-30T13:00:00.000');
+    fixture.detectChanges();
+    expect(fixture.nativeElement.innerText).not.toEqual('23h59min ago');
+    expect(fixture.nativeElement.innerText).not.toEqual('24h0min ago');
+  });
+
+
+  it('shows "yesterday at HH:mm" for dates between 24 hours ago and yesterday beginning of day', () => {
+    const fixture = TestBed.createComponent(Timestamp);
+    const componentInstance = fixture.componentInstance;
+
+    componentInstance.date = new Date('2020-06-30T13:00:00.000');
+    fixture.detectChanges();
+    expect(fixture.nativeElement.innerText).toEqual('yesterday at 13:00');
+
+    componentInstance.date = new Date('2020-06-30T08:23:23.000');
+    fixture.detectChanges();
+    expect(fixture.nativeElement.innerText).toEqual('yesterday at 08:23');
+
+    componentInstance.date = new Date('2020-06-30T00:01:00.000');
+    fixture.detectChanges();
+    expect(fixture.nativeElement.innerText).not.toEqual('yesterday at 00:01');
+
+    componentInstance.date = new Date('2020-06-30T00:00:00.000');
+    fixture.detectChanges();
+    expect(fixture.nativeElement.innerText).not.toEqual('yesterday at 00:00');
+  });
+
+  it('shows absolute timestamp for dates older than yesterday beginning of day', () => {
+    const fixture = TestBed.createComponent(Timestamp);
+    const componentInstance = fixture.componentInstance;
+
+    let date = new Date('2020-06-29T23:59:59.999');
+    componentInstance.date = date;
+    fixture.detectChanges();
+    expect(fixture.nativeElement.innerText)
+      .toEqual(DateTime.fromJSDate(date).toLocaleString(DateTime.DATETIME_MED));
```

Comment:
As a reader, I don't know what output to expect and need to read up on the luxon documentation. If possible, prefer to test against a string literal (`.toEqual('yesterday at 00:00')`) and not a function output (`toLocaleString(DateTime.DATETIME_MED)`).

Reply:
I was thinking that it may be different across localizations, but for test purposes I suppose it won't be a problem.